lib/algfld.gd

```diff
@@ -188,3 +188,45 @@ DeclareGlobalFunction("AlgExtEmbeddedPol");
 
 DeclareGlobalFunction("AlgExtSquareHensel");
 
+#############################################################################
+##
+#F  DecomPoly( <f> [:"onlyone"] )  finds ideal decompositions of rational f
+##
+##  <#GAPDoc Label="DecomPoly">
+##  <ManSection>
+##  <Func Name="DecomPoly" Arg='pol'/>
+##
+##  <Description>
+##  Let <M>f</M> be a univariate, rational, irreducible, polynomial. A
+##  pair <M>g</M>,<M>h</M> of polynomials of degree strictly
+##  smaller than that of <M>f</M>, such that <M>f(x)|g(h(x))</M> is
+##  called an ideal decomposition. In the context of field
+##  extensions, if <M>\alpha</M> is a root of <M>f</M> in a suitable extension
+##  and <M>Q</M> the field of rational numbers, such a decomposition corresponds
+##  to (proper) subfields <M>Q\lt Q(\beta)\lt Q(\alpha)</M>, where <M>g</M> is the
+##  minimal polynomial of <M>\beta</M>.
+##  This function determines such decompositions up to equality of the subfields
+##  <M>Q(\beta)</M>, thus determining subfields of a given algebraic extension.
+##  It returns a list of pairs <M>[g,h]</M> (and an empty list if no such
+##  decomposition exists). If the option <A>onlyone</A> is given it returns at
+##  most one such decomposition (and performs faster).
+##  <Example><![CDATA[
+##  gap> x:=X(Rationals,"x");;pol:=x^8-24*x^6+144*x^4-288*x^2+144;;
```

Comment:
Offtopic: the reference manual says this on `X`:
> X is simply a synonym for Indeterminate. However, we do not recommend to use this synonym which is supported only for the backwards compatibility.

Now, don't get me wrong, I am not suggesting you should change the examples; rather, I always use `X` myself, so I am wondering if this last sentence in the manual should just be deleted?

Reply:
Yes. (At the time when this was written there was a push to make `X` obsolete as operation.)